lib/datadog/tracing/span_operation.rb

```diff
@@ -439,6 +443,11 @@ def message
       # mutation by reference; when this span is returned,
       # we don't want this SpanOperation to modify it further.
       def build_span
+        # Use parent service if no service is set and @inherit_parent_service is set
+        service = self.service || (@parent&.service if @inherit_parent_service)
+
+        service_entry = @parent.nil? || (service && @parent.service != service)
```

Comment:
What is `@parent`? It concerns me that `SpanOperation` has a reference to `parent`, because of the deep coupling/nesting... more references means less garbage collection means more memory bloat. It also makes it harder to move data structures like this off the critical path to a background process, if there was an optimization opportunity to do so.

I know my original intention was to eliminate `@parent` altogether. But I can't remember if we couldn't and if I just had to minimize its usage. I'm not hot on the idea of expanding its usage...